plugins/graphql-common/src/createLoader.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { GraphQLContext } from '@backstage/plugin-graphql-common';
+import DataLoader, { Options } from 'dataloader';
+import { GraphQLError } from 'graphql';
+import { BatchLoadFn } from './types';
+
+/** @public */
+export const createLoader = <Context extends GraphQLContext>(
+  loaders: Record<string, BatchLoadFn<Context>>,
+  options?: Options<string, any>,
+) => {
+  return (context: Context) => {
+    async function fetch(ids: readonly string[]) {
+      const { decodeId } = context;
```

Comment:
Wondering if this should simply be a fixed string format? Is there benefit to having the provided and potentially configurable?

Reply:
Yeah, let's use RFC suggestion